spec/requests/graphql/mutations/update_user_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+module Mutations
+  module Users
+    RSpec.describe UpdateUser, type: :request do
+      describe '.resolve' do
+        it 'updates a user' do
+          frosty = User.create!(name: 'Frosty', email: 'snow@brr.com', birthdate: DateTime.new(2001, 2, 3))
+
+          post graphql_path, params: { query: query(frosty.id) }
+          result = JSON.parse(response.body)
+
+          data = result['data']['updateUser']
+
+          expect(data['name']).to eq('Frosted Flakes Frosty')
+          expect(data['email']).to eq('ice@brr.com')
+          expect(data['birthdate']).to eq('2000-01-01 00:00:00 UTC')
+        end
+
+        def query(user_id)
+          <<~GQL
+            mutation {
+              updateUser(
+                id: #{user_id}
+                name: "Frosted Flakes Frosty"
+                email: "ice@brr.com"
+                birthdate: "2000-01-01"
+                ) {
+                id
+                name
+                email
+                birthdate
+                }
+            }
+          GQL
```

Comment:
For the mutation query, you have to explicitly ask for the return fields in lines 29-32 in order to get them returned back. By default the mutation only returns the user id after it performs the mutation.

I think there may be a way to get it to return all those field by default which may make our queries a bit dryer? Or maybe it goes against the GraphQL idea of "you get exactly what you ask for". Thoughts?

Reply:
Yeah I think the perk of GraphQL is being able to only request what you want! So I think explicitly calling for the returns is what we want.

Reply:
Nice request spec here.

Reply:
Great test Angela!